### PR TITLE
fix: kill-child-process

### PR DESCRIPTION
**Pull request by oneofthezombies**
# PR Template

## Checklist

- [ ] E2E finished.
- [ ] Local, Self Hosted and Development environment test finished.
- [x] Local, E2E, Self Hosted, Development and Production env variable checked.
- [x] (Backend only) DB migration file creation checked.
- [x] **If this is a API deleted or changed**: Check side effects for API deletion or related code change.

### CODE REVIEW

File: packages/typescript/node/package.json [hunk 1]
```diff
@@ -35,6 +35,7 @@
     "plist": "3.0.6",
     "reflect-metadata": "0.1.13",
     "rxjs": "7.5.6",
+    "tree-kill": "^1.2.2",
```
Comment (yowpark) [inline packages/typescript/node/package.json]: caret issue

**Comment by yowpark** [comment]
Thank you so much

File: packages/typescript/node/src/tree-kill.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { ChildProcess } from 'child_process';
+import treeKill from 'tree-kill';
+
+export { treeKill };
+
+export function killChildProcess(childProcess: ChildProcess, signal: string | number = 'SIGTERM'): Promise<void> {
+  return new Promise((resolve, reject) => {
+    if (childProcess.pid === undefined || childProcess.exitCode !== null || childProcess.signalCode !== null) {
```
[the remark this answers is not in the document]
Comment (oneofthezombies) [reply packages/typescript/node/src/tree-kill.ts:8]: I checked that exitCode or signalCode had a value only when the child process exited.
After running the java server process with `spawn()`,
The result values are as follows when the kill command is delivered to `0, SIGINT, SIGTERM, SIGKILL, and SIGCONT`.
Therefore, signalCode does not have a value "**when process receive the signal**" but have a value "**when process receive the signal and exit**". 🔥🔥🔥

| signal  | exitCode | signalCode |
| ------- | -------- | ---------- |
| SIGINT  | 130      | null       |
| SIGTERM | 143      | null       |
| SIGKILL | null     | SIGKILL    |
| 0       | null     | null       |
| SIGCONT | null     | null       |

[the remark this answers is not in the document]
Comment (yowpark) [reply packages/typescript/node/src/tree-kill.ts:8]: Nice test!. thanks for information

**Pull request merged**